File: WEEK 1/C++/Dairy Farm Profit Calculation.cpp
#include <iostream>
using namespace std;

int main() {
    const double MilkCartonVolumeLiters = 3.78;
    const double CostPerLiter = 0.38;
    const double ProfitPerCarton = 0.27;

    cout << "Enter the total amount of milk produced in the morning (liters): ";
    double totalMilkProduced;

    
    while (true) {
        cin >> totalMilkProduced;
        if (totalMilkProduced >= 0.0) {
            break;  
        } else {
            cout << "Invalid input. Please enter a non-negative value: ";
        }
    }

    int numberOfCartons = static_cast<int>(totalMilkProduced / MilkCartonVolumeLiters);
    double costOfProduction = totalMilkProduced * CostPerLiter;
    double profit = numberOfCartons * ProfitPerCarton;

    printf("Number of milk cartons needed: %d\n", numberOfCartons);
    printf("Cost of producing the milk: $%.2f\n", costOfProduction);
    printf("Profit for producing the milk: $%.2f\n", profit);

    return 0;
}

File: WEEK 1/C#/Dairy Farm Profit Calculation.cs
using System;

class Program {
    static void Main() {
        const decimal MilkCartonVolumeLiters = 3.78m;
        const decimal CostPerLiter = 0.38m;
        const decimal ProfitPerCarton = 0.27m;

        Console.Write("Enter the total amount of milk produced in the morning (liters): ");
        decimal totalMilkProduced = decimal.Parse(Console.ReadLine());

        int numberOfCartons = (int)(totalMilkProduced / MilkCartonVolumeLiters);
        decimal costOfProduction = totalMilkProduced * CostPerLiter;
        decimal profit = numberOfCartons * ProfitPerCarton;

        Console.WriteLine($"Number of milk cartons needed: {numberOfCartons}");
        Console.WriteLine($"Cost of producing the milk: ${costOfProduction:F2}");
        Console.WriteLine($"Profit for producing the milk: ${profit:F2}");
    }
}
